aspnetcore/fundamentals/http-requests/samples/Startup.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using HttpClientFactorySample.GitHub;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace HttpClientFactorySample
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMemoryCache();
+
+            // basic usage
+            services.AddHttpClient();
+
+            // named client
+            services.AddHttpClient("github", c =>
+            {
+                c.BaseAddress = new Uri("https://api.github.com/");
+
+                c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // Github API versioning
+                c.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // Github requires a user-agent
+            });
+
+            // configuration occurs in ctor
+            services.AddHttpClient<GitHubService>();
```

Comment:
@stevejgordon do you have a preference between these two styles of registration? Ryan and I have talked about it in the past and I am looking for a third opinion on it :).

Reply:
@glennc I lean towards the ctor configuration personally. I find in larger real-world apps, the ConfigureServices method can get quite complex, even if things end up split out into extension methods. Putting the configuration into the typed client feels closer to where I'd personally expect to look for it.